rust/chains/hyperlane-ethereum/src/merkle_tree_hook.rs

```diff
@@ -230,53 +242,35 @@ where
     #[instrument(skip(self))]
     #[allow(clippy::needless_range_loop)]
     async fn tree(&self, maybe_lag: Option<NonZeroU64>) -> ChainResult<IncrementalMerkle> {
-        let lag = maybe_lag.map(|v| v.get()).unwrap_or(0).into();
-
-        let fixed_block_number: BlockNumber = self
-            .provider
-            .get_block_number()
-            .await
-            .map_err(ChainCommunicationError::from_other)?
-            .saturating_sub(lag)
-            .into();
+        let mut call = self.contract.tree();
+        if let Some(lag) = maybe_lag {
```

Comment:
wonder if it's possible to easily dedupe this logic? like a function that takes in a generic call and the `maybe_lag` and then applies this logic